test/pack/pack.c

```diff
@@ -220,8 +220,13 @@ void *runtest(void *arg)
                 rc = DTP_obj_buf_init(sobj, sbuf_h, 0, 1, basecount);
                 assert(rc == DTP_SUCCESS);
 
-                rc = DTP_obj_buf_init(dobj, dbuf_h, -1, 0, basecount);
-                assert(rc == DTP_SUCCESS);
+                if (strstr(typestr, "complex") != NULL) {
+                    rc = DTP_obj_buf_init(dobj, dbuf_h, 0, 0, basecount);
+                    assert(rc == DTP_SUCCESS);
+                } else {
+                    rc = DTP_obj_buf_init(dobj, dbuf_h, -1, 0, basecount);
+                    assert(rc == DTP_SUCCESS);
+                }
```

Comment:
So previously the dest buffer was not initialized?

Anyway, I don't think this commit belongs in this PR.

Reply:
The way complex is initialized, it is hard to calculate the correct product of complex. This fix is to initialize complex array to a different value so that we can easily do result validation. This is not perfect because assigning complex to zero over-simplifies it. This PR is a collection of recent commits for various things, all commits are not related to a theme.